domains/measurements/Measurements.API/Tests/Fixtures/TestServerFixture.cs

```diff
@@ -0,0 +1,229 @@
+        {
+            EnsureServer();
+
+            return GrpcChannel.ForAddress(_server!.BaseAddress, new GrpcChannelOptions
+            {
+                LoggerFactory = LoggerFactory,
+                HttpHandler = _handler
+            });
+        }
+
+        public HttpClient CreateUnauthenticatedClient(string environment = "Development")
+        {
+            EnsureServer(environment);
+
+            var client = _server!.CreateClient();
+            client.DefaultRequestHeaders.Add("EO_API_VERSION", "20240110");
+            return client;
+        }
+
+        public HttpClient CreateAuthenticatedClient(string sub, string tin = "11223344", string name = "Peter Producent",
+            string actor = "d4f32241-442c-4043-8795-a4e6bf574e7f", string apiVersion = "20240110", string environment = "Development")
+        {
+            EnsureServer(environment);
+            var client = _server!.CreateClient();
+            client.DefaultRequestHeaders.Authorization =
+                new AuthenticationHeaderValue("Bearer", GenerateToken(sub: sub, tin: tin, name: name, actor: actor));
+            client.DefaultRequestHeaders.Add("EO_API_VERSION", apiVersion);
+            return client;
+        }
+
+        private string GenerateToken(
+            string scope = "",
+            string actor = "d4f32241-442c-4043-8795-a4e6bf574e7f",
+            string sub = "03bad0af-caeb-46e8-809c-1d35a5863bc7",
+            string tin = "11223344",
+            string cpn = "Producent A/S",
+            string name = "Peter Producent",
+            string issuer = "Issuer",
+            string audience = "Audience")
+        {
+            var claims = new Dictionary<string, object>()
+            {
+                { UserClaimName.Scope, scope },
+                { UserClaimName.ActorLegacy, actor },
+                { UserClaimName.Actor, actor },
+                { UserClaimName.Tin, tin },
+                { UserClaimName.OrganizationName, cpn },
+                { JwtRegisteredClaimNames.Name, name },
+                { UserClaimName.ProviderType, ProviderType.MitIdProfessional.ToString()},
+                { UserClaimName.AllowCprLookup, "false"},
+                { UserClaimName.AccessToken, ""},
+                { UserClaimName.IdentityToken, ""},
+                { UserClaimName.ProviderKeys, ""},
+                { UserClaimName.OrganizationId, sub},
+                { UserClaimName.MatchedRoles, ""},
+                { UserClaimName.Roles, ""},
+                { UserClaimName.AssignedRoles, ""}
+            };
+
+            var signedJwtToken = new TokenSigner(PrivateKey).Sign(
+                sub,
+                name,
+                issuer,
+                audience,
+                null,
+                60,
+                claims
+            );
+
+            return signedJwtToken;
+        }
+
+        protected virtual void Dispose(bool disposing)
+        {
+            if (!_disposed)
```

Comment:
Why these if statements? cant we just dispose when the IDisposable.Dispose is hit?

Reply:
Not sure - again copy paste